modules/packages/LinearAlgebra.chpl

```diff
@@ -396,19 +396,24 @@ proc _array.T where isDefaultRectangularArr(this) && this.domain.rank == 2
   return transpose(this);
 }
 
-/* Add matrices, maintaining dimensions, deprecated for ``_array.plus`` */
+/*Add matrices, maintaining dimensions. Preferred method at scale */
 proc matPlus(A: [?Adom] ?eltType, B: [?Bdom] eltType) where isDefaultRectangularArr(A) && isDefaultRectangularArr(B) {
-  compilerWarning('matPlus has been deprecated for _array.plus, ' +
-                  'try: A.plus(B)');
+  compilerWarning('matMinus has been deprecated for _array.plus, ' +
+                  'try: A.minus(B)');
   return A.plus(B);
 }
 
 /* Add matrices, maintaining dimensions */
-proc _array.plus(A: [?Adom]) where isDefaultRectangularArr(A) && isDefaultRectangularArr(this) {
-  if Adom.rank != this.domain.rank then compilerError("Unmatched ranks");
-  if Adom.shape != this.domain.shape then halt("Unmatched shapes");
-  var C: [Adom] eltType = this + A;
-  return C;
+proc _array.plus(A: [?Adom]) where isCSArr(A) && isCSArr(this) {
+  if this.domain.parentDom != Adom.parentDom then halt("Unmatched Shapes");
+  var sps = CSRDomain(Adom.parentDom);
+  sps += this.domain;
+  sps += Adom;
+  var S: [sps] real;
```

Comment:
We'd like this to be generic w.r.t. element type, so we should refer to the array `eltType` here. You can do this cleanly by adding an element type query to the signature, and use that type variable here, e.g.

```chpl
proc _array.plus(A: [?Adom] ?eltType) where isCSArr(A) && isCSArr(this) {

...

var S: [sps] eltType;
```

Reply:
ah good point